STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/AclCommandSupport.java

```diff
@@ -0,0 +1,205 @@
+            repo = deployConfigRepo;
+        }
+        String resource = split[2].trim();
+        if (repo == null && !resource.isEmpty()) {
+            throw new CommandFormatException(MSG4);
+        }
+        if (Action.ADD == action || Action.REMOVE == action || Action.SET == action) {
+            SidType sidType = toSidType(split[3].trim());
+            String sid = split[4].trim();
+            String[] permissions = split[5].trim().split(",");
+            permissions = Arrays.stream(permissions)
+                .map(String::trim)
+                .map(String::toUpperCase)
+                .filter(e -> !e.isEmpty())
+                .toArray(String[]::new);
+            if (permissions.length == 0 && Action.SET != action) {
+                throw new CommandFormatException(MSG2);
+            }
+            for (String permission : permissions) {
+                if (listAllSupportedPermissions(repoType).stream()
+                    .noneMatch(e -> Objects.equals(AclPermission.toString(e), permission.toUpperCase()))) {
+                    throw new CommandFormatException(String.format(MSG3, permission, repoType.getName()));
+                }
+            }
+            return new AclCommand(action, sidType, sid, repoType, repo, resource, permissions);
+        } else {
+            return new AclCommand(action, null, null, repoType, repo, resource, null);
+        }
+    }
+
+    private static String toRepo(String repo) {
+        int index = repo.indexOf("/");
+        if (index > 0) {
+            repo = repo.substring(index + 1).trim();
+        } else {
+            return null;
+        }
+        if (repo.length() == 0) {
+            return null;
+        }
+        return repo;
+    }
+
+    private static Action toAction(String command) throws CommandFormatException {
+        if ("add".equalsIgnoreCase(command)) {
+            return Action.ADD;
+        } else if ("set".equalsIgnoreCase(command)) {
+            return Action.SET;
+        } else if ("remove".equalsIgnoreCase(command)) {
+            return Action.REMOVE;
+        } else if ("list".equalsIgnoreCase(command)) {
+            return Action.LIST;
+        } else if ("listAll".equalsIgnoreCase(command)) {
+            return Action.LIST_ALL;
+        } else {
+            throw new CommandFormatException(String.format(MSG5, command));
+        }
+    }
+
+    private static SidType toSidType(String sidType) throws CommandFormatException {
+        switch (sidType.toLowerCase()) {
+            case "user":
+                return SidType.USERNAME;
+            case "group":
+                return SidType.GROUP_NAME;
+            case "groupid":
+                return SidType.GROUP_ID;
+            default:
+                throw new CommandFormatException(String.format(MSG6, sidType));
+        }
+    }
+
+    private static RepoType toRepoType(String type) throws CommandFormatException {
+        int index = type.indexOf("/");
+        if (index > 0) {
```

Comment:
Is it OK for "/repo" or just "/"?

Reply:
Yep, the pattern "design/design_id" is used. 0 is skipped.

Reply:
I mean, is it possible to define "/design_id" or "/" or "design/"? Is there checks on this case?